sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ExtRefEditorService.java

```diff
@@ -488,9 +489,24 @@ public void updateIedNameBasedOnLnode(SCL scl) {
                                                                 && Objects.equals(tlNode.getLnInst(), tCompasFlow.getExtReflnInst())
                                                                 && Utils.lnClassEquals(tlNode.getLnClass(), tCompasFlow.getExtReflnClass())
                                                                 && Objects.equals(tlNode.getPrefix(), tCompasFlow.getExtRefprefix()))
+                                                        .filter(tlNode -> {
```

Comment:
why do you use another filter ?

```java
.filter(tlNode -> {
TCompasICDHeader tCompasICDHeader = PrivateUtils.extractCompasPrivate(tlNode, TCompasICDHeader.class)
.orElseThrow(() -> new ScdException(("The substation LNode with following attributes : IedName:"+tlNode.getIedName()+" / LdInst:"+tlNode.getLdInst()+" / LnClass:"+tlNode.getLnClass().getFirst()+" / LnInst:"+tlNode.getLnInst()+"  does not contain the needed private (COMPAS - ICDHeader)")));
    return Objects.equals(tlNode.getLdInst(), tCompasFlow.getExtRefldinst())
            && Objects.equals(tlNode.getLnInst(), tCompasFlow.getExtReflnInst())
            && Utils.lnClassEquals(tlNode.getLnClass(), tCompasFlow.getExtReflnClass())
            && Objects.equals(tlNode.getPrefix(), tCompasFlow.getExtRefprefix())
            && Objects.equals(tCompasFlow.getFlowSourceIEDType(), tCompasICDHeader.getIEDType())
            && Objects.equals(tCompasFlow.getFlowIEDSystemVersioninstance(), tCompasICDHeader.getIEDSystemVersioninstance())
            && Objects.equals(tCompasFlow.getFlowSourceIEDredundancy(), tCompasICDHeader.getIEDredundancy());
})
```

Reply:
I think this is less explicit and does not correspond to the issue I will keep it like this.